src/tracking.ts

```diff
@@ -0,0 +1,222 @@
+import { getItem, RequestError, setItem } from ".";
+
+export const STORAGE_KEY = "searchio_events";
+const EXPIRY_DAYS = 30;
+const FUNNEL_ENTRY_TYPES = ["click"];
+
+type Metadata = Record<string, boolean | number | string>;
+interface EventState {
+  queryId: string;
+  type: string;
+  timestamp: number;
+  submitted: boolean;
+  metadata?: Metadata;
+}
+type Value = string | number;
+type Identifier = "result_id" | "redirect_id" | "promotion_id";
+
+export class SearchIOAnalytics {
```

Comment:
We auto-generate documentation based on docblocks. You can do that later if you wish.

Reply:
I'll add that now.